code/modules/hydroponics/hydro_tray.dm

```diff
@@ -916,32 +471,67 @@
 
 	update_icon()
 
-/obj/machinery/portable_atmospherics/hydroponics/soil
-	name = "soil"
-	icon = 'icons/obj/hydroponics.dmi'
-	icon_state = "soil"
-	density = 0
-	use_power = 0
-	draw_warnings = 0
+/obj/machinery/portable_atmospherics/hydroponics/verb/set_label()
+	set name = "Set Tray Label"
+	set category = "Object"
+	set src in view(1)
 
-/obj/machinery/portable_atmospherics/hydroponics/soil/attackby(var/obj/item/O as obj, var/mob/user as mob)
-	if(istype(O, /obj/item/weapon/pickaxe/shovel))
-		if(!seed)
-			user << "You clear up [src]!"
-			new /obj/item/weapon/ore/glass(loc)//we get some of the dirt back
-			new /obj/item/weapon/ore/glass(loc)
-			qdel(src)
-		else
-			..()
-	else if(istype(O,/obj/item/weapon/pickaxe/shovel) || istype(O,/obj/item/weapon/tank))
+	if(!usr || usr.stat || usr.restrained() || (usr.status_flags & FAKEDEATH))
 		return
-	else
-		..()
 
-/obj/machinery/portable_atmospherics/hydroponics/soil/New()
+	var/n_label = input(usr, "What would you like to set the tray's label display to?", "Hydroponics Tray Labeling", null)  as text
```

Comment:
We really, really, really need an universal proc for this kind of sanitizing. There's probably 40 fucking different ways that people have tried to sanitize inputs like these over the years.